src/fundus/publishers/base_objects.py

```diff
@@ -59,8 +59,8 @@ def __init__(self, spec: PublisherSpec):
         for url_source in spec.sources:
             if not isinstance(url_source, URLSource):
                 raise TypeError(
-                    f"Unexpected type '{type(url_source).__name__}' as source for {self.name}. "
-                    f"Allowed are '{', '.join(cls.__name__ for cls in iterate_all_subclasses(URLSource))}'"
+                    f"Unexpected type {type(url_source).__name__!r} as source for {self!r}. "
+                    f"Allowed are {', '.join(repr(cls.__name__) for cls in iterate_all_subclasses(URLSource))}"
```

Comment:
If there is no special reason to do it this way, I would suggest, for consistencies sake, to do it this way: `', '.join(cls.__name__ for ... )!r`

Reply:
Those two produce different outcomes and the current one is the desired.